src/style-spec/validate_mapbox_api_supported.js

```diff
@@ -106,7 +106,8 @@ function getRootErrors(style: Object, specKeys: Array<any>): Array<?ValidationEr
         'draft',
         'created',
         'modified',
-        'visibility'
+        'visibility',
+        'protected'
```

Comment:
Can you update the comment above optionalRootProperties to describe how the Style API interacts with protected?